TestModels/Resource/codegen-patches/rust/dafny-4.5.0.patch

```diff
@@ -0,0 +1,782 @@
++++ a/TestModels/Resource/runtimes/rust/dafny_impl/src/implementation_from_dafny.rs
+@@ -548,7 +548,7 @@ pub mod r#_simple_dresources_dinternaldafny_dtypes {
+             }
+         }
+     }
+-
++/*
+     impl ::std::default::Default for GetResourcesOutput {
+         fn default() -> GetResourcesOutput {
+             GetResourcesOutput::GetResourcesOutput {
+@@ -556,7 +556,7 @@ pub mod r#_simple_dresources_dinternaldafny_dtypes {
+             }
+         }
+     }
+-
++*/
+     impl ::std::convert::AsRef<GetResourcesOutput> for &GetResourcesOutput {
+         fn as_ref(&self) -> Self {
+             self
+@@ -588,7 +588,7 @@ pub mod r#_simple_dresources_dinternaldafny_dtypes {
+             let mut output = ::dafny_runtime::MaybePlacebo::<::std::rc::Rc<super::r#_Wrappers_Compile::Result<::std::rc::Rc<super::r#_simple_dresources_dinternaldafny_dtypes::GetResourceDataOutput>, ::std::rc::Rc<super::r#_simple_dresources_dinternaldafny_dtypes::Error>>>>::new();
+             let mut _out0 = ::dafny_runtime::MaybePlacebo::<::std::rc::Rc<super::r#_Wrappers_Compile::Result<::std::rc::Rc<super::r#_simple_dresources_dinternaldafny_dtypes::GetResourceDataOutput>, ::std::rc::Rc<super::r#_simple_dresources_dinternaldafny_dtypes::Error>>>>::new();
+             _out0 = ::dafny_runtime::MaybePlacebo::from(
+-                ::dafny_runtime::md!(self).r#_GetResourceData_k(input),
++                self.r#_GetResourceData_k(input)
+             );
+             output = ::dafny_runtime::MaybePlacebo::from(_out0.read());
+             return output.read();
+@@ -845,13 +845,14 @@ pub mod r#_SimpleResource_Compile {
+                 ::std::rc::Rc<super::r#_simple_dresources_dinternaldafny_dtypes::Error>,
+             >,
+         > {
+-            let mut _out1 = ::dafny_runtime::MaybePlacebo::<::std::rc::Rc<super::r#_Wrappers_Compile::Result<::std::rc::Rc<super::r#_simple_dresources_dinternaldafny_dtypes::GetResourceDataOutput>, ::std::rc::Rc<super::r#_simple_dresources_dinternaldafny_dtypes::Error>>>>::new();
+-            _out1 = ::dafny_runtime::MaybePlacebo::from(
++            // let mut _out1 = ::dafny_runtime::MaybePlacebo::<::std::rc::Rc<super::r#_Wrappers_Compile::Result<::std::rc::Rc<super::r#_simple_dresources_dinternaldafny_dtypes::GetResourceDataOutput>, ::std::rc::Rc<super::r#_simple_dresources_dinternaldafny_dtypes::Error>>>>::new();
++            let _out1 = ::dafny_runtime::MaybePlacebo::from(
+                 super::r#_simple_dresources_dinternaldafny_dtypes::ISimpleResource::GetResourceData(
+-                    self.unwrap(),
++                    self,
+                     input,
+                 ),
+             );
++            _out1.read()
+         }
+         pub fn _ctor(
+             this: &::dafny_runtime::Object<Self>,
+@@ -958,7 +959,6 @@ pub mod r#_SimpleResource_Compile {
+                 },
+             ));
+             return output.read();
+-            return output.read();
```

Comment:
The double return is a known issue that will be fixed soon at least, so reasonable to fix it. :) Was it a compile error?